### PR TITLE
Mingw fixes

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Motivation for this change

Some non-mass rebuild fixes for mingw.

- unmark a few packages as unix only
- fixes builds of:
  - file
  - openssl
  - ncurses
- add ```libgnurx``` (needed for file)
- add two wine tests to verify executables work

### CODE REVIEW

File: pkgs/data/fonts/dejavu-fonts/default.nix [hunk 1]
```diff
@@ -19,8 +19,6 @@ let
     # DejaVu changes are in public domain
     # See http://dejavu-fonts.org/wiki/License for details
     license = stdenv.lib.licenses.free;
-
-    platforms = stdenv.lib.platforms.unix;
```
Comment (dezgeg) [inline pkgs/data/fonts/dejavu-fonts/default.nix]: Set it to `all` instead of deleting or it won't be built on Hydra.

**Comment by matthewbauer** [comment]
@grahamcofborg build tests.cross.mingw32 tests.cross.mingwW64 pkgsCross.mingw32.openssl pkgsCross.mingwW64.openssl pkgsCross.mingw32.libxslt pkgsCross.mingwW64.libxslt pkgsCross.mingw32.ncurses pkgsCross.mingwW64.ncurses pkgsCross.mingw32.gettext pkgsCross.mingwW64.gettext

**Comment by matthewbauer** [comment]
@GrahamcOfBorg build tests.cross.mingw32 tests.cross.mingwW64 pkgsCross.mingw32.openssl pkgsCross.mingwW64.openssl pkgsCross.mingw32.libxslt pkgsCross.mingwW64.libxslt pkgsCross.mingw32.ncurses pkgsCross.mingwW64.ncurses pkgsCross.mingw32.gettext pkgsCross.mingwW64.gettext

**Comment by matthewbauer** [comment]
Dropped the gettext one, everything else seems to build!

Eventually this test should go in release-cross.nix.

**Comment by doronbehar** [comment]
Hey @matthewbauer , hope you don't mind me resurrecting a (super) old PR. Do you manage to cross compile `file` to windows?

```
$ nix build -Lf. pkgsCross.mingw32.file
file-i686-w64-mingw32> /nix/store/3six896mnb1qxgc2k3aqaznnw0bqm3ip-i686-w64-mingw32-binutils-2.40/bin/i686-w64-mingw32-ld: .libs/funcs.o: in function `file_pipe_closexec':
file-i686-w64-mingw32> /build/file-5.44/src/funcs.c:884: undefined reference to `pipe'
file-i686-w64-mingw32> collect2: error: ld returned 1 exit status
file-i686-w64-mingw32> make[3]: *** [Makefile:500: libmagic.la] Error 1
file-i686-w64-mingw32> make[3]: Leaving directory '/build/file-5.44/src'
file-i686-w64-mingw32> make[2]: *** [Makefile:381: all] Error 2
file-i686-w64-mingw32> make[2]: Leaving directory '/build/file-5.44/src'
file-i686-w64-mingw32> make[1]: *** [Makefile:462: all-recursive] Error 1
file-i686-w64-mingw32> make[1]: Leaving directory '/build/file-5.44'
file-i686-w64-mingw32> make: *** [Makefile:371: all] Error 2
```

**Pull request merged**